File: test/rs/case06_string.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{math_cos, math_exp, math_floor, math_sin, math_sqrt, perf_counter, py_bool, py_grayscale_palette, py_in, py_isalpha, py_isdigit, py_len, py_print, py_save_gif, py_slice, py_write_rgb_png};

// このファイルは自動生成です（native Rust mode）。

fn greet(name: &String) -> String {
    return format!("{}{}", "Hello, ".to_string(), name);
}

fn main() {
    py_print(greet(&("Codex".to_string())));
}

File: test/cpp/case06_string.cpp
// このファイルは `test/py/case06_string.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。


#include "cpp_module/py_runtime.h"

// このファイルは `test/py/case06_string.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

str greet(const str& name) {
    return "Hello, " + name;
}

int main() {
    py_print(greet("Codex"));
    return 0;
}
